RFCs/FS-1148-units.md

```diff
@@ -0,0 +1,74 @@
+# F# RFC FS-1148 - Ease conversion between Units of Measure (UoM) and undecorated numerals and simplify casting
+
+The design suggestion [Ease conversion between Units of Measure (UoM) and undecorated numerals and simplify casting](https://github.com/fsharp/fslang-suggestions/issues/892) has been marked "approved in principle".
+
+This RFC covers the detailed proposal for this suggestion.
+
+- [x] [Suggestion](https://github.com/fsharp/fslang-suggestions/issues/892)
+- [x] Approved in principle
+- [x] [Implementation](https://github.com/dotnet/fsharp/pull/17518)
+- [ ] Discussion
+
+## Summary
+
+A new type `Units` will be added to FSharp.Core and provide `static` methods for addition, removal and casting of UoM for supported primitive types and common collections containing primitive types.
+
+## Motivation
+
+Prior to this RFC, F# programmers have been able to apply units of measure supported primitive types through the use of functions in the `LanguagePrimitives` module e.g. `LanguagePrimitives.FloatWithMeasure<'M>`, or, by multiplying the value by one `1.0<kg>`. It has been commonly asked by the community for a simple way to remove units of measure from a given value or a way to convert units of measure from one unit to another directly. It has also been requested that there should be a way to convert collections of primitives e.g. `int[]` to `int<kg>[]` without having to create a copy of the collection.
+
+## Detailed design
+
+### FSharp.Core additions
+
+The additions follow a common pattern:
+
+- An overloaded `Add` method for each of the supported primitive types adds a UoM to the value
+- An overloaded `Remove` method for each of the supported primitive types removed UoM from the value
+- An overloaded `Cast` method for each of the supported primitive types replaces UoM on the value
+- Overloaded `Add*`, `Remove*` and `Cast*` e.g. `AddArray` for common collection types containing supported primitives:
+  - `Array`
+  - `List`
+  - `ResizeArray`
+  - `Seq`
+
+Total added methods is 195 `13 supported primitives * 3 methods * (primitive + 4 collection types)`. They are presented as 15 methods with 13 overloads each.
+
+```fsharp
+namespace Microsoft.FSharp.Core
+
+
+type Units =
```

Comment:
The purpose of units of measure (including in F#) is to add safety when doing arithmetic and interfacing between systems which may have different conventions. Sometimes measures need to be removed or added when interfacing with systems that do not understand measures. This is inevitable and these are always potential points of failure. To preserve safety here we must have `[<RequiresExplicitTypeArguments>]` on all of these. It is then explicit in the code what measure is being added or removed. Without this, the wrong measure (e.g. `cm` instead of `m`) may get added or removed by mistake, leading to bugs (e.g. numbers wrong by a factor of 100).